From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Thu, 20 Jan 2011 17:44:28 +0000 (+0100)
Subject: ASoC: ABE DSP: Correct mute/unmute to manage Left/Right channels
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=07d51e09af259060a5a1ac95f7d5db67e5fbed68;hp=dea73678cf13aa7834f757299d892638eb3e88bd

ASoC: ABE DSP: Correct mute/unmute to manage Left/Right channels

Left and right channels should be control for BE mute/unmute ports.
Today only one channel is controlled.

Change-Id: If583ea45f87c6b9588f7e6bc131cfbe016ba2c6e
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---

Index: android-2.6.35/sound/soc/omap/omap-abe.c
===================================================================
--- android-2.6.35.orig/sound/soc/omap/omap-abe.c	2011-01-24 16:39:08.000000000 +0530
+++ android-2.6.35/sound/soc/omap/omap-abe.c	2011-01-25 11:28:53.644092202 +0530
@@ -852,20 +852,24 @@
 
 		switch (be_rtd->dai_link->be_id) {
 		case OMAP_ABE_DAI_PDM_UL:
-			abe_mute_gain(GAINS_AMIC, 0);
+			abe_mute_gain(GAINS_AMIC, GAIN_LEFT_OFFSET);
+			abe_mute_gain(GAINS_AMIC, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_BT_VX:
 		case OMAP_ABE_DAI_MM_FM:
 		case OMAP_ABE_DAI_MODEM:
 			break;
 		case OMAP_ABE_DAI_DMIC0:
-			abe_mute_gain(GAINS_DMIC1, 0);
+			abe_mute_gain(GAINS_DMIC1, GAIN_LEFT_OFFSET);
+			abe_mute_gain(GAINS_DMIC1, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_DMIC1:
-			abe_mute_gain(GAINS_DMIC2, 0);
+			abe_mute_gain(GAINS_DMIC2, GAIN_LEFT_OFFSET);
+			abe_mute_gain(GAINS_DMIC2, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_DMIC2:
-			abe_mute_gain(GAINS_DMIC3, 0);
+			abe_mute_gain(GAINS_DMIC3, GAIN_LEFT_OFFSET);
+			abe_mute_gain(GAINS_DMIC3, GAIN_RIGHT_OFFSET);
 			break;
 		}
 	}
@@ -885,20 +889,24 @@
 
 		switch (be_rtd->dai_link->be_id) {
 		case OMAP_ABE_DAI_PDM_UL:
-			abe_unmute_gain(GAINS_AMIC, 0);
+			abe_unmute_gain(GAINS_AMIC, GAIN_LEFT_OFFSET);
+			abe_unmute_gain(GAINS_AMIC, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_BT_VX:
 		case OMAP_ABE_DAI_MM_FM:
 		case OMAP_ABE_DAI_MODEM:
 			break;
 		case OMAP_ABE_DAI_DMIC0:
-			abe_unmute_gain(GAINS_DMIC1, 0);
+			abe_unmute_gain(GAINS_DMIC1, GAIN_LEFT_OFFSET);
+			abe_unmute_gain(GAINS_DMIC1, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_DMIC1:
-			abe_unmute_gain(GAINS_DMIC2, 0);
+			abe_unmute_gain(GAINS_DMIC2, GAIN_LEFT_OFFSET);
+			abe_unmute_gain(GAINS_DMIC2, GAIN_RIGHT_OFFSET);
 			break;
 		case OMAP_ABE_DAI_DMIC2:
-			abe_unmute_gain(GAINS_DMIC3, 0);
+			abe_unmute_gain(GAINS_DMIC3, GAIN_LEFT_OFFSET);
+			abe_unmute_gain(GAINS_DMIC3, GAIN_RIGHT_OFFSET);
 			break;
 		}
 	}
